keel-core/src/main/kotlin/com/netflix/spinnaker/keel/pause/ActuationPauser.kt

```diff
@@ -102,10 +102,35 @@ class ActuationPauser(
     mutableListOf<PersistentEvent>()
       .let { events ->
         events.addAll(originalEvents)
-        val relevantAppEvents = resourceRepository
-          .applicationEventHistory(resource.application, events.last().timestamp)
+
+        val oldestResourceEvent = events.last()
+
+        if (oldestResourceEvent is ResourceCreated) {
```

Comment:
This is much simplified from my previous attempt. Now that we have timestamps associated with records in the `paused` table (which are not deleted upon deleting delivery configs or resources), we can synthesize paused events in a more straightforward fashion.